### DISCUSSION TITLE
Test to pass if a specific exit code

### DISCUSSION THREAD

**Discussion by dgibbs64** [community/community#25954] [Actions]
<p>When running some tests and am actually checking for specific exit codes to ensure it fails with the exit code expected.</p>
<p>For example, if the code exits with exit code 1 then the test has passed.</p>
<p>However with Actions when this happens then it is classed as a failure.</p>
<p>Is there a way to state if a specific exitcode is output then it is a pass.?</p>
<p>Thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That would be possible by wrapping your test in something that checks the exit code, e.g. a Python script:</p>
<pre><code class="lang-python">import subprocess
import sys
p = subprocess.run(command)
p.returncode != 1:
    print(f'Unexpected return code from {command}: {p.returncode}', file=sys.stderr)
    sys.exit(1)
</code></pre>
<p>However, you should only do that if you’re specifically testing error conditions. A successful run of your program should always lead to an error code of zero.</p>
<p>If you want, you can integrate the Python script into the workflow file by setting <code>shell: python</code> on a <code>run</code> step.</p>

**Comment by dgibbs64** [comment]
<p>Thanks for this. It is the sort of thing that I thought I might have to do. It is a shame Actions doesn’t have something built-in yet.</p>
<p>For the majority of tests, I am looking for a 0 exit code, so this would only apply for some tests where I specifically want a failure.</p>
<p>My script only outputs the following 0 (pass), 1 (fatal), 2 (error), 3 (warn) and depending upon the test I will be looking for one of these.</p>
<p>My scripts are all BASH so I will likely create what you have posted in BASH.</p>
<p>Maybe a wrapper script that I can pass parameters that will then run the script and if the script provides the expected exit code with the wrapper will output exit code 0 and exit code 1 for unexpected code.</p>
<p>e.g</p>
<pre><code class="lang-auto">./exitcode.sh [command] [exitcode]
./exitcode.sh "./script.sh" "1"
</code></pre>
<p>Hopefully, there will be an inbuilt option in the future. But for now, I think this should do the trick.</p>
